model/riscv_insts_zcmp.sail

```diff
@@ -0,0 +1,206 @@
+/*=======================================================================================*/
+/*  This Sail RISC-V architecture model, comprising all files and                        */
+/*  directories except where otherwise noted is subject the BSD                          */
+/*  two-clause license in the LICENSE file.                                              */
+/*                                                                                       */
+/*  SPDX-License-Identifier: BSD-2-Clause                                                */
+/*=======================================================================================*/
+
+function clause currentlyEnabled(Ext_Zcmp) = hartSupports(Ext_Zcmp) & currentlyEnabled(Ext_Zca) & not(currentlyEnabled(Ext_Zcd))
+
+mapping zcmp_assembly_rlist : bits(4) <-> string = {
+  0x4 <-> "ra",
+  0x5 <-> "ra" ^ sep() ^ "s0",
+  0x6 <-> "ra" ^ sep() ^ "s0-s1",
+  0x7 <-> "ra" ^ sep() ^ "s0-s2",
+  0x8 <-> "ra" ^ sep() ^ "s0-s3",
+  0x9 <-> "ra" ^ sep() ^ "s0-s4",
+  0xa <-> "ra" ^ sep() ^ "s0-s5",
+  0xb <-> "ra" ^ sep() ^ "s0-s6",
+  0xc <-> "ra" ^ sep() ^ "s0-s7",
+  0xd <-> "ra" ^ sep() ^ "s0-s8",
+  0xe <-> "ra" ^ sep() ^ "s0-s9",
+  0xf <-> "ra" ^ sep() ^ "s0-s11",
+  // Using X names instead of ABI names.
```

Comment:
The compiler should really warn about overlapping mappings.  A better approach might be the way `reg_name` works by checking `get_config_use_abi_names()` and then calling the underlying `raw` mappings for X and ABI names.

Reply:
How about
```
mapping zcmp_assembly_rlist : bits(4) <-> string = {
  i if get_config_use_abi_names()      <-> zcmp_abi_assembly_rlist(i),
  i if not(get_config_use_abi_names()) <-> zmp_arch_assembly_rlist(i),
}
```
where the `_abi_` and `_arch_` functions do the appropriate mapping.  Each function will be shorter and the `get_config_use_abi_names()` check doesn't have to be repeated on each line.